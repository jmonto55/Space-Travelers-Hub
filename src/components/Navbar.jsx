import { Link, useLocation } from 'react-router-dom';
import rocket from '../assets/rocket.svg';

const Navbar = () => {
  const location = useLocation();
  return (
    <div className="max-w-screen-2xl m-auto w-full">
      <nav className="container  mx-auto flex py-5 justify-between items-center">
        <div className="flex justify-center items-center">
          <img className="w-14" src={rocket} alt="space hub logo" />
          <h1 className="ml-5 text-3xl font-semibold">Space Travelers Hub</h1>
        </div>
        <ul className="flex gap-10">
          <li>
            <Link className={`font-semibold text-lg hover:text-slate-600 text-blue-500 ${location.pathname === '/' ? 'underline' : ''}`} to="/">Rockets</Link>
          </li>
          <li>
            <Link className={`font-semibold text-lg hover:text-slate-600 text-blue-500 ${location.pathname === '/missions' ? 'underline' : ''}`} to="missions">Missions</Link>
          </li>
          <li className="border-r-2 border-black h-4 mt-2" />
          <li>
            <Link className={`font-semibold text-lg hover:text-slate-600 text-blue-500 ${location.pathname === '/myprofile' ? 'underline' : ''}`} to="myprofile">My Profile</Link>
          </li>
        </ul>
      </nav>
      <div className="flex justify-center px-12">
        <hr className="w-full" />
      </div>
    </div>
  );
};

export default Navbar;
