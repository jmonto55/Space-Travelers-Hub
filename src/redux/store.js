import { configureStore, combineReducers } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionsSlice';

const rootReducer = combineReducers({ rockets: rocketsReducer, missions: missionReducer });

const store = configureStore({ reducer: rootReducer });

export default store;
