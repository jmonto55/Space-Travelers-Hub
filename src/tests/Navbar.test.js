import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Navbar from './MockNavbar';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Navbar component', () => {
  it('Should render a Snapshot of the Navbar component', () => {
    const navbar = render(
      <Provider store={store}>
        <Navbar />
      </Provider>,
    );
    expect(navbar).toMatchSnapshot();
  });

  test('Should test if the texts "Rockets, Missions & Space Travelers Hub" are inside the Navbar component', () => {
    render(
      <Provider store={store}>
        <Navbar />
      </Provider>,
    );
    expect(screen.getByText('Rockets' && 'Missions' && 'Space Travelers Hub')).toBeInTheDocument();
  });
});
