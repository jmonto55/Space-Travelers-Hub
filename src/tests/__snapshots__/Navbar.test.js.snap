// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Navbar component Should render a Snapshot of the Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-screen-2xl m-auto w-full"
      >
        <nav
          class="flex px-20 py-5 justify-between items-center"
        >
          <div
            class="flex justify-center items-center"
          >
            <img
              alt="space hub logo"
              class="w-14"
              src="rocket.svg"
            />
            <h1
              class="ml-5 text-3xl font-semibold"
            >
              Space Travelers Hub
            </h1>
          </div>
          <ul
            class="flex gap-10"
          >
            <li>
              Rockets
            </li>
            <li>
              Missions
            </li>
            <li>
              My Profile
            </li>
          </ul>
        </nav>
        <div
          class="flex justify-center px-12"
        >
          <hr
            class="w-full"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-screen-2xl m-auto w-full"
    >
      <nav
        class="flex px-20 py-5 justify-between items-center"
      >
        <div
          class="flex justify-center items-center"
        >
          <img
            alt="space hub logo"
            class="w-14"
            src="rocket.svg"
          />
          <h1
            class="ml-5 text-3xl font-semibold"
          >
            Space Travelers Hub
          </h1>
        </div>
        <ul
          class="flex gap-10"
        >
          <li>
            Rockets
          </li>
          <li>
            Missions
          </li>
          <li>
            My Profile
          </li>
        </ul>
      </nav>
      <div
        class="flex justify-center px-12"
      >
        <hr
          class="w-full"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
